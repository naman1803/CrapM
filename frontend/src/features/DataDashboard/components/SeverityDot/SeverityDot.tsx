import React from 'react';

export interface SeverityDotProps {
    cx: number;
    cy: number;
    severityLevel: string;
}

const SeverityDot: React.FC<SeverityDotProps> = ({ cx, cy, severityLevel }) => {
    switch (severityLevel) {
        case "Low":
            return (
                <image
                    x={cx - 20}
                    y={cy - 20}
                    width={'40'}
                    height={'40'}
                    href="https://bluemoji.io/cdn-proxy/646218c67da47160c64a84d5/66b3e5d0c2ab246786ca1d5e_86.png"
                />
            );
        case "Moderate":
            return (
                <image
                    x={cx - 20}
                    y={cy - 20}
                    width={'40'}
                    height={'40'}
                    href="https://bluemoji.io/cdn-proxy/646218c67da47160c64a84d5/66b3e58bf4a1d9b1b453e5df_91.png"
                />
            );
        case "High":
            return (
                <image
                    x={cx - 20}
                    y={cy - 20}
                    width={'40'}
                    height={'40'}
                    href="https://bluemoji.io/cdn-proxy/646218c67da47160c64a84d5/66b3e67f60bd0c05aa29a1d6_70.png"
                />
            );
        case "Critical":
        default:
            return (
                <image
                    x={cx - 20}
                    y={cy - 20}
                    width={'40'}
                    height={'40'}
                    href="https://bluemoji.io/cdn-proxy/646218c67da47160c64a84d5/66b3e5760608f3f68cb0b04b_93.png"
                />
            );
    }
};

export default SeverityDot;