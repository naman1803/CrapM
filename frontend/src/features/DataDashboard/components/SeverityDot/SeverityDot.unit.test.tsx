import { render } from '@testing-library/react';
import SeverityDot, { SeverityDotProps } from './SeverityDot';

describe('SeverityDot', () => {
    const defaultProps: SeverityDotProps = {
        cx: 50,
        cy: 50,
        severityLevel: 'Low',
    };

    it('renders correctly with severity level "Low"', () => {
        const { asFragment } = render(<SeverityDot {...defaultProps} severityLevel="Low" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with severity level "Moderate"', () => {
        const { asFragment } = render(<SeverityDot {...defaultProps} severityLevel="Moderate" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with severity level "High"', () => {
        const { asFragment } = render(<SeverityDot {...defaultProps} severityLevel="High" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with severity level "Critical"', () => {
        const { asFragment } = render(<SeverityDot {...defaultProps} severityLevel="Critical" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
