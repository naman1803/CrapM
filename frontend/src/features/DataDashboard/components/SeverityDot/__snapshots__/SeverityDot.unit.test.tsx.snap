// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeverityDot renders correctly with severity level "Critical" 1`] = `
<DocumentFragment>
  <img
    height="40"
    href="https://bluemoji.io/cdn-proxy/646218c67da47160c64a84d5/66b3e5760608f3f68cb0b04b_93.png"
    width="40"
    x="30"
    y="30"
  />
</DocumentFragment>
`;

exports[`SeverityDot renders correctly with severity level "High" 1`] = `
<DocumentFragment>
  <img
    height="40"
    href="https://bluemoji.io/cdn-proxy/646218c67da47160c64a84d5/66b3e67f60bd0c05aa29a1d6_70.png"
    width="40"
    x="30"
    y="30"
  />
</DocumentFragment>
`;

exports[`SeverityDot renders correctly with severity level "Low" 1`] = `
<DocumentFragment>
  <img
    height="40"
    href="https://bluemoji.io/cdn-proxy/646218c67da47160c64a84d5/66b3e5d0c2ab246786ca1d5e_86.png"
    width="40"
    x="30"
    y="30"
  />
</DocumentFragment>
`;

exports[`SeverityDot renders correctly with severity level "Moderate" 1`] = `
<DocumentFragment>
  <img
    height="40"
    href="https://bluemoji.io/cdn-proxy/646218c67da47160c64a84d5/66b3e58bf4a1d9b1b453e5df_91.png"
    width="40"
    x="30"
    y="30"
  />
</DocumentFragment>
`;
